lib/class-wp-rest-themes-controller.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * REST API: WP_REST_Themes_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.0.0
+ */
+
+/**
+ * Core class used to manage themes via the REST API.
+ *
+ * @since 5.0.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Themes_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.0.0
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'themes';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/active',
```

Comment:
Per [discussion in Slack](https://wordpress.slack.com/archives/C02RQC26G/p1539359220000100), we should use the following structure for querying this information:
- `/wp/v2/themes?status=active` will return an array containing the object representing the capabilities of the active theme.
- For this initial release, `status` will be an `enum` parameter with one option, `active`, and that parameter will be required.
  - This leaves us room to expand the functionality of the endpoint to permit listing all themes down the road, etc, but restricts the complexity to the bare minimum currently needed.
  - Using a filter further avoids any lock-in around how to point to the active theme from other resources (the debate between a link from the index, `/active`, `/~active`, etc, which has unfolded mostly in slack). Any of these approaches might be found to be limiting down the road as we add support for theme management, multi-site, _etcetera_, while a query parameter like `status` is unlikely to conflict with any future plans.

Long-term I imagine we'd update the active theme by PUT'ing a theme slug to the settings endpoint, or a hypothetical new site endpoint. Figuring all that out is out of scope and not necessary for Gutenberg in terms of 5.0.